pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -769,15 +769,24 @@ func (n *NetworkPolicyController) processNetworkPolicy(np *networkingv1.NetworkP
 		}
 	}
 
+	// Set EnableLogging field for the given rule.
+	configureEnableLogging := func(originalRule controlplane.NetworkPolicyRule) controlplane.NetworkPolicyRule {
+		logConfiguredRule := originalRule
+		if enableLogging {
+			logConfiguredRule.EnableLogging = enableLogging
+		}
+		return logConfiguredRule
+	}
+
 	// If ingress isolation is specified explicitly and there's no ingress rule, append a deny-all ingress rule.
 	// See https://kubernetes.io/docs/concepts/services-networking/network-policies/#default-deny-all-ingress-traffic
 	if ingressIsolated && !ingressRuleExists {
-		rules = append(rules, denyAllIngressRule)
+		rules = append(rules, configureEnableLogging(denyAllIngressRule))
```

Comment:
it might be simpler to just create the rule:
```suggestion
		rules = append(rules, controlplane.NetworkPolicyRule{
			Direction: controlplane.DirectionIn,
			EnableLogging: enableLogging,
		})
```
or create a func to construct a deny all rule for readability:
```
// denyAllRule returns a NetworkPolicyRule which denies all traffic in the given direction.
func denyAllRule(direction controlplane.Direction, enableLogging bool) controlplane.NetworkPolicyRule {
	return &controlplane.NetworkPolicyRule{
		Direction: controlplane.DirectionIn,
		EnableLogging: enableLogging,
	}
}
```

Reply:
Sure, happy to create the rule directly. I adopted the func method to keep exclusive this manual inserted rule.